bin/tests/it/k8s.rs

```diff
@@ -1000,20 +1086,18 @@ fn get_agent_ds_yaml(
                 map
             })
             .collect();
-        if let Some(serde_json::Value::Array(containers)) = cfgmap.get_mut("containers") {
-            if let Some(serde_json::Value::Object(container)) = containers.get_mut(0) {
-                if let Some(serde_json::Value::Array(existing_env_vars)) = container.get_mut("env")
-                {
-                    for env_var in env_vars {
-                        let map: serde_json::Map<String, serde_json::Value> = env_var
-                            .into_iter()
-                            .map(|(k, v)| (k.to_string(), v))
-                            .collect();
-                        existing_env_vars.push(serde_json::Value::Object(map));
-                    }
-                } else {
-                    panic!("unexpected");
-                }
+        let existing_env_vars = &mut cfgmap["spec"]["template"]["spec"]["containers"][0]["env"]
+            .as_array_mut()
+            .unwrap();
+        for env_var in env_vars {
+            if let Some((k, v)) = env_var.iter().next() {
```

Comment:
nit: I find the previous method of converting `env_var` into a map and then collecting it to be a bit more ergonomic, but by no means is this bad/wrong/blocker.

Reply:
test code. explicit easy to debug.